packit/base_git.py

```diff
@@ -0,0 +1,95 @@
+import git
+import logging
+from typing import Optional
+
+from packit.config import Config, PackageConfig
+from packit.exceptions import PackitException
+from packit.local_project import LocalProject
+
+logger = logging.getLogger(__name__)
+
+
+class BaseGitClass:
+    def __init__(self, config: Config, package_config: PackageConfig):
+        self.config = config
+        self.package_config = package_config
+        self.package_name: Optional[str] = self.package_config.downstream_package_name
+        self.dist_git_url: Optional[str] = self.package_config.downstream_project_url
+        logger.debug(f"Using dist-git repo {self.dist_git_url}")
+        self._local_project = None
+        self.upstream_project_url: str = self.package_config.upstream_project_url
+        self._specfile = None
+
+    def create_branch(
+        self,
+        local_project: LocalProject,
```

Comment:
I would prefer to use `local_project` from the `Class`. (If the common implementation is not possible, create an abstract property in this class and implement for each `SubClass` independently.)